ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OnDisk.hs

```diff
@@ -1313,23 +1339,57 @@ sm secParam db = StateMachine {
     , shrinker      = shrinker
     , semantics     = semantics db
     , mock          = symbolicResp
-    , cleanup       = noCleanup
+    , cleanup       = \_ -> do
+        LedgerBackingStore bs <- atomically . readTVar $ dbBackingStore db
+        HD.bsClose bs
+        dbCleanup $ dbEnv db
     }
 
-prop_sequential :: SecurityParam -> QC.Property
-prop_sequential secParam = QC.withMaxSuccess 100000 $
-    forAllCommands (sm secParam dbUnused) Nothing $ \cmds ->
-      QC.monadicIO (propCmds secParam cmds)
-
+prop_sequential :: (SecurityParam -> IO (DbEnv IO)) -> SecurityParam -> QC.Property
```

Comment:
I believe we use `prop_sequential` to test for arbitrary values of `SecurityParam` at lines 101-107, so we can't omit it here. We also can't remove it from `DbEnv IO` because `initStandaloneDB` depends on it. We could test for arbitrary values of `DbEnv IO` that contain arbitrary values of `SecurityParam`, but I don't think we can come up with a sensible `Arbitrary` instance for it. Maybe we can still make it a bit nicer with minor refactoring?